selfdrive/updated.py

```diff
@@ -303,11 +305,36 @@ def attempt_update():
 
     finalize_from_ovfs_copy()
 
+    # If a NEOS update is required, download it in the background
+    with open("/VERSION", "r") as current_neos_file:
+      current_neos_version = current_neos_file.read().replace("\n", "")
+    required_neos_version = run(["bash", "-c", 
+               r"unset REQUIRED_NEOS_VERSION && source launch_env.sh && echo -n $REQUIRED_NEOS_VERSION"],
+               FINALIZED)
+    print(f"NEOS version update check: {current_neos_version} current, {required_neos_version} in update")
+    if current_neos_version != required_neos_version or DEBUG_FORCE_UPDATE:
+      print(f"Beginning background download for NEOS {required_neos_version}")
+      update_json = f'file:///{FINALIZED}/installer/updater/update.json'
+      while True:
```

Comment:
I would add a maximum number of attempts here

Reply:
If the user leaves Internet coverage, the maximum number of attempts at 10 minutes per would have to be really high to avoid unnecessarily restarting the entire process.

Reply:
Think we still want to bail out after like a day / 144 attempts in case we mess something up and ship a broken manifest or something.